docs/essentials/extending.md

```diff
@@ -0,0 +1,141 @@
+:::
+
+::: warning
+Native support for extensions is enabled for files generated by `IocConcept` (all files imported from `#ioc` directory).
+:::
+
+## What are extensions used to?
+
+As it was said at the beginning, extensions can be used to extend functionality of existing files. For example you have a mapper in your common-module and you want to add an extra key mapping in your custom-module, extension is ideal solution to your problem.
+
+### Example of using extensions in mappers concept
+
+Let's imagine, you have a `ToProduct.ts` mapper (in a module-a), which is used to properly map back-end response to your key/value pairs.
+
+```ts
+// module-a/mappers/ToProduct.ts
+
+import ToProduct from '#ioc/mappers/ToProduct'
+import ToMoney from '#ioc/mappers/ToMoney'
+
+export default (data: any) => ({
+  __typename: data.__typename ?? '',
+  id: (data.id ?? 0) as number,
+  sku: (data.sku ?? '') as string,
+  name: (data.name ?? '') as string,
+  finalPrice: ToMoney(data.price_range?.minimum_price?.final_price ?? {}),
+  crossSellProducts: (data.related_products ?? []).map(ToProduct),
+  // ...
+})
+```
+
+::: info
+As we can see, the original file exports a function which takes one parameter (data) and returns a mapped object.
+:::
+
+We want to add to that key/value pairs with our module-b. So, we create the same concept (folder structure) and the same file (with `.ext.ts` suffix) in a new module.
+
+```ts
+// module-b/mappers/ToProduct.ext.ts
+
+import ToProductLabel from '#ioc/mappers/ToProductLabel'
+import Extension from '#ioc/types/base/Extension'
+
+interface Labels {
+  labels: ReturnType<typeof ToProductLabel>[]
+}
+
+const ToProduct: Extension<Labels> = (ToProduct) => (data) => {
+  const product = ToProduct(data)
+
+  product.labels = data.product_labels?.items.map(ToProductLabel) ?? []
+
+  return product
+}
+
+export default ToProduct
+```
+
+As you can see the extension is a function, which takes one parameter `ToProduct` that is the original mapper and returns a function which gets one parameter `data` that is the data recieved by mapper. Inside we can do some stuff with original object (add `labels` property). Similar to the original mapper, mapped object is returned at the end.
+
+### Example of using extensions in services concept
+
+In this example, we will try to solve the following issue, that is to add some functionality to existing composable. You have a `useLoginCustomer.ts` service inside a `services` concept. This service is a composable, so it returns a function inside the exported function. Its purpose is to logIn a user and set a cookie for him. You can see the example service bellow.
+
+```ts
+// customer-magento/services/useLoginCustomer.ts
+
+import useLoginCustomerRepository from '#ioc/repositories/useLoginCustomerRepository'
+import useCookies from '#ioc/composables/useCookies'
+import MAGENTO_CUSTOMER_COOKIE_NAME from '#ioc/config/MAGENTO_CUSTOMER_COOKIE_NAME'
```

Comment:
Only problem is this no longer exists 😀 

There are now useCustomerToken and useCartToken composables for manipulating with these cookies.